portfolio/src/main/webapp/game.js

```diff
@@ -100,10 +117,106 @@ class Board {
       } else {
         this.board[row][col] = 2;
       }
+      this.numTiles++;
+      this.checkGameState(this.redTrue);
       this.redTrue = !this.redTrue;
+    } else {
+        alert("Cannot place tile there. Try again.");
+    }
+  }
+  
+  /*
+  * Checks whether move made by redTrue player resulted in win or if game board is full.
+  * Sends alert with win message or gameOver message if board is full and allows user to
+  * restart game.  
+  */
+  checkGameState(redTrue) {
+    var player = 0;
+    if (redTrue) {
+      player = 1; 
+    } else {
+        player = 2;
+    }
+    var win = false; 
+
+    // check rows
+    for (var i = 0; i < this.board.length; i++) {
+      var rowWin = true;
+      for (var j = 0; j < this.board[i].length; j++) {
+        if (this.board[i][j] != player) {
+          rowWin = false;
+        }
+      }
+      if (rowWin) {
+        win = true;
+        break;
+      }
+    }
+
+    // check cols 
+    for (var j = 0; j < this.board[0].length; j++) {
+      var colWin = true;
+      for (var i = 0; i < this.board.length; i++) {
+        if (this.board[i][j] != player) {
```

Comment:
In JS we use `===` & `!==` to match type as well as value.

Behavior:
'2' == 2 // True
'2' === 2 // False